main.go

```diff
@@ -170,6 +177,8 @@ func main() {
 		}
 		exitOnError(instaScaleController.SetupWithManager(context.Background(), mgr), "Error setting up InstaScale controller")
 	}
+	rayClusterController := cfoControllers.RayClusterReconciler{Client: mgr.GetClient(), Scheme: mgr.GetScheme()}
```

Comment:
Let's add the controller conditionally to the presence of the RayCluster API on the cluster?

Reply:
On second thought, can you address my comment above?:
> This can cause race conditions when both components are enabled. If CFO starts before KubeRay, then this Reconciler will never start despite the API being present soon after

Reply:
@astefanutti Can you take a look at the conditional I've added for this? I was having trouble finding other examples and I'm not sure the way I did it is best.